src/ORM/DataList.php

```diff
@@ -1361,15 +1366,14 @@ private function fetchEagerLoadManyMany(
         $joinRows = [];
         if (!empty($parentIDs) && !empty($fetchedIDs)) {
             $fetchedIDsAsString = implode(',', $fetchedIDs);
-            $joinRows = DB::query(
-                'SELECT * FROM "' . $joinTable
-                // Only get joins relevant for the parent list
-                . '" WHERE "' . $parentIDField . '" IN (' . implode(',', $parentIDs) . ')'
-                // Exclude any children that got filtered out
-                . ' AND ' . $childIDField . ' IN (' . $fetchedIDsAsString . ')'
-                // Respect sort order of fetched items
-                . ' ORDER BY FIELD(' . $childIDField . ', ' . $fetchedIDsAsString . ')'
-            );
+            $joinRows = SQLSelect::create()
+                ->setSelect('"' . $joinTable . '".' . "*")
+                ->setFrom([$joinTable => $joinTable])
+                ->addWhere('"' . $parentIDField . '" IN (' . implode(',', $parentIDs) . ')')
+                ->addWhere('"' . $childIDField . '" IN (' . $fetchedIDsAsString . ')')
+                ->addLeftJoin($childTable, "$childTable.ID = $joinTable.$childIDField")
```

Comment:
Let's say we have a joinTable AB linking A's many_many to B. The original query preserved the order of the items that were fetched from table B by putting their IDs in the FIELD list. This changes that by joining table B so that the original order by used to fetch B items can also be used in this query.

The perfect solution would be to right join the join table to the relation query (line 1350), which removes the need for this query altogether, though I'm not sure what the implications will be. I'll have to look into that.